src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -806,6 +817,27 @@ public static ServerInfo GetServerVersion(IDbConnection connection)
                     serverInfo.OsVersion = reader[0].ToString();
                 });
 
+                // Get BDC endpoints
+                if (serverInfo != null && serverInfo.Options != null && serverInfo.Options.ContainsKey(ServerInfo.OptionIsBigDataCluster))
+                {
+                    int? isBigDataCluster = serverInfo.Options[ServerInfo.OptionIsBigDataCluster] as int?;
+                    if (isBigDataCluster != null && isBigDataCluster.Value == 1)
+                    {
+                        ExecuteReader(
+                        connection,
+                        SqlConnectionHelperScripts.GetClusterEndpoints,
+                        delegate (IDataReader reader)
+                        {
+                            List<ClusterEndpoint> clusterEndpoints = new List<ClusterEndpoint>();
```

Comment:
Minor: what happens if we don't have permissions to this table / can't find it? In the future, this is a possibility. We should consider this.